src/ui/tabline.rs

```diff
@@ -82,9 +82,35 @@ impl Tabline {
 
         glib::signal_handler_block(&self.notebook, &self.switch_tab_signal);
 
+        let mut nvim = nvim.lock().unwrap();
+
         let mut page = 0;
         for (i, tab) in tabs.iter().enumerate() {
-            let tab_label = gtk::Label::new(tab.1.as_str());
```

Comment:
What do you think of extracting the logic to a function that receives `Tabpage, Arc<Mutex<Neovim>>` and returns the `tab_label`?

Reply:
Good idea, I like that better than keeping all the logic in the for loop. (It would make the logic’s meaning more clear too.) What do you think of naming it `get_tab_label` and putting it above `update`? Let me know and I can push the change.